client/state/data-getters/index.js

```diff
@@ -82,6 +82,47 @@ export const requestGeoLocation = () =>
 		{ fromApi: () => ( { body: { country_short } } ) => [ [ 'geo', country_short ] ] }
 	);
 
+export const requestRecentPostViews = ( siteId, postIds ) => {
+	const key = `recent-post-views-${ siteId }`;
+	const freshness = 5 * 60 * 1000; // 5 min
+
+	// Don't re-request recent post views that are cached and up to date OR
+	// have a pending request.
+	const freshPostIds = postIds.filter(
+		postId =>
+			getHttpData( `${ key }-${ postId }` ).lastUpdated >= Date.now() - freshness &&
+			getHttpData( `${ key }-${ postId }` ).status !== 'pending'
+	);
+	const requestPostIds = difference( postIds, freshPostIds );
```

Comment:
at this point even though I suggested the `difference()` it was because I hastily copied from above. since we're already filtering this line isn't necessary and neither is the import

I believe that if we wanted the same behavior we could rearrange the filter. writing it out in English can help…

> we want to only fetch those post counts which…
>  - are in the list of requested post ids
>  - are not already pending an API call
>  - are not already in state fresher than our needs

```js
const postIdsToRefresh = postIds.filter( postId => {
	const cached = getHttpData( `${ key }-${ postId }` );

	return (
		( cached.status !== 'pending' ) &&
		( cached.lastUpdated < Date.now() - freshness )
	);
} );
```